components/activities/activities-tab/activities-tab.tsx

```diff
@@ -118,20 +93,54 @@ export function ActivitiesTabManager<
         <div className="list-header">
           ACTIVITY HISTORY
         </div>
-        {pastActivities?.length
-          ? (
-            <div className="list-group">
-              {pastActivities?.map((activity) => {
-                const ActivityTile = activity.sourceType
-                  ? activityListItems[activity.sourceType]
-                  : null;
-                return ActivityTile &&
-                  React.createElement(ActivityTile, { activity: activity.activity });
-              })}
-            </div>
-          )
-          : <div className="empty-list-group">No past activities</div>}
+        {/* activity li's will only rerender if their data changes */}
+        <Memo>
+          {() => (
+            <ActivityGroup
+              activityConnection$={activityAlertConnection$}
+              activityListItems={activityListItems}
+              isActive={false}
+              emptyStateMessage="No past activities"
+            />
+          )}
+        </Memo>
       </div>
     </div>
   );
 }
+
+function ActivityGroup<
+  ActivityListItemTypes,
+  SourceType extends StringKeys<ActivityListItemTypes>,
+  ActivityType extends ActivityListItemTypes[SourceType],
+>(
+  { activityConnection$, activityListItems, isActive, emptyStateMessage }: {
+    activityConnection$: Observable<ActivityConnection<ActivityType, SourceType>>;
```

Comment:
can we call the type Signal?

Reply:
We could but I think it would be nice to just use the legend types since there are a load of different built-in types w/ the observable prefix and it could be confusing if we exported the high level `Observable` type to `Signal` and all of the other legend types used Observable.

https://github.com/LegendApp/legend-state/blob/bf25e657a0cbfd5f9a52a87b710a26a40904761d/src/observableInterfaces.ts